scripts/dts/gen_defines.py

```diff
@@ -277,669 +172,97 @@ def relativize(path):
-        out(f"INST_{instance_no}_{str2ident(compat)}", 1)
-
-
-def node_ident(node):
-    # Returns an identifier for 'node'. Used e.g. when building macro names.
-
-    # TODO: Handle PWM on STM
-    # TODO: Better document the rules of how we generate things
-
-    ident = ""
-
-    if node.bus_node:
-        ident += "{}_{:X}_".format(
-            str2ident(node.bus_node.matching_compat), node.bus_node.unit_addr)
-
-    ident += f"{str2ident(node.matching_compat)}_"
-
-    if node.unit_addr is not None:
-        ident += f"{node.unit_addr:X}"
-    elif node.parent.unit_addr is not None:
-        ident += f"{node.parent.unit_addr:X}_{str2ident(node.name)}"
-    else:
-        # This is a bit of a hack
-        ident += str2ident(node.name)
-
-    return ident
-
-
-def write_addr_size(edt, prop_name, prefix):
-    # Writes <prefix>_BASE_ADDRESS and <prefix>_SIZE for the node pointed at by
-    # the /chosen property named 'prop_name', if it exists
-
-    node = edt.chosen_node(prop_name)
-    if not node:
-        return
-
-    if not node.regs:
-        err("missing 'reg' property in node pointed at by "
-            f"/chosen/{prop_name} ({node!r})")
-
-    out_comment(f"/chosen/{prop_name} ({node.path})")
-    out(f"{prefix}_BASE_ADDRESS", hex(node.regs[0].addr))
-    out(f"{prefix}_SIZE", node.regs[0].size//1024)
-
-
-def write_flash(edt):
-    # Writes chosen and tree-wide flash-related output
-
-    write_flash_node(edt)
-    write_code_partition(edt)
-
-    if flash_area_num != 0:
-        out_comment("Number of flash partitions")
-        out("FLASH_AREA_NUM", flash_area_num)
-
-
-def write_flash_node(edt):
-    # Writes output for the top-level flash node pointed at by
-    # zephyr,flash in /chosen
-
-    node = edt.chosen_node("zephyr,flash")
-
-    out_comment(f"/chosen/zephyr,flash ({node.path if node else 'missing'})")
-
-    if not node:
-        # No flash node. Write dummy values.
-        out("FLASH_BASE_ADDRESS", 0)
-        out("FLASH_SIZE", 0)
-        return
-
-    if len(node.regs) != 1:
-        err("expected zephyr,flash to have a single register, has "
-            f"{len(node.regs)}")
-
-    if node.on_bus == "spi" and len(node.bus_node.regs) == 2:
-        reg = node.bus_node.regs[1]  # QSPI flash
-    else:
-        reg = node.regs[0]
-
-    out("FLASH_BASE_ADDRESS", hex(reg.addr))
-    if reg.size:
-        out("FLASH_SIZE", reg.size//1024)
-
-    if "erase-block-size" in node.props:
-        out("FLASH_ERASE_BLOCK_SIZE", node.props["erase-block-size"].val)
-
-    if "write-block-size" in node.props:
-        out("FLASH_WRITE_BLOCK_SIZE", node.props["write-block-size"].val)
-
+        try:
+            instance_no = node.edt.compat2enabled[compat].index(node)
+        except ValueError:
+            continue
+        idents.append(f"N_INST_{instance_no}_{str2ident(compat)}")
+    # Node labels
+    idents.extend(f"N_NODELABEL_{str2ident(label)}" for label in node.labels)
```

Comment:
> (`idents += [f"N_NODELABEL_{str2ident(label)}" for label in node.labels]` works too.)

Since they both work and there's no reason to change this, I'm leaving it as is.